src/core/Akka.Tests/Pattern/BackoffOnRestartSupervisorSpec.cs

```diff
@@ -0,0 +1,339 @@
+
+                Receive<string>(str => str.Equals("THROW"), msg =>
+                {
+                    throw new NormalException();
+                    return;
+                });
+
+                Receive<string>(str => str.Equals("THROW_STOPPING_EXCEPTION"), msg =>
+                {
+                    throw new StoppingException();
+                    return;
+                });
+
+                Receive<Tuple<string, string>>(str => str.Item1.Equals("TO_PARENT"), msg =>
+                {
+                    Context.Parent.Tell(msg.Item2);
+                });
+
+                ReceiveAny(other => _probe.Tell(other));
+            }
+
+            public static Props Props(IActorRef probe)
+            {
+                return Akka.Actor.Props.Create(() => new TestActor(probe));
+            }
+        }
+
+        public class TestParentActor : ReceiveActor
+        {
+            private readonly IActorRef _probe;
+
+            public TestParentActor(IActorRef probe, Props supervisorProps)
+            {
+                _probe = probe;
+                var supervisor = Context.ActorOf(supervisorProps);
+
+                ReceiveAny(other => _probe.Forward(other));
+            }
+
+            public static Props Props(IActorRef probe, Props supervisorProps)
+            {
+                return Akka.Actor.Props.Create(() => new TestParentActor(probe, supervisorProps));
+            }
+        }
+
+        public class SlowlyFailingActor : ReceiveActor
+        {
+            private readonly TestLatch _latch;
+
+            public SlowlyFailingActor(TestLatch latch)
+            {
+                _latch = latch;
+
+                Receive<string>(str => str.Equals("THROW"), msg =>
+                {
+                    Sender.Tell("THROWN");
+                    throw new NormalException();
+                    return;
+                });
+
+                Receive<string>(str => str.Equals("PING"), msg =>
+                {
+                    Sender.Tell("PONG");
+                });
+            }
+
+            protected override void PostStop()
+            {
+                _latch.Ready(3.Seconds());
+            }
+
+            public static Props Props(TestLatch latch)
+            {
+                return Akka.Actor.Props.Create(() => new SlowlyFailingActor(latch));
+            }
+        }
+
+        private Props SupervisorProps(IActorRef probeRef)
+        {
+            var options = Backoff.OnFailure(TestActor.Props(probeRef), "someChildName", 200.Milliseconds(), 10.Seconds(), 0.0)
+                .WithSupervisorStrategy(new OneForOneStrategy(4, TimeSpan.FromSeconds(30), ex =>
+                {
+                    if (ex is StoppingException)
+                    {
+                        return Directive.Stop;
+                    }
+                    // TODO: should restart be there?
+                    return Directive.Restart;
```

Comment:
Because that's the default directive and I think they're able to express that through an `implicit`. You can remove the comment.